### ISSUE TITLE
Modifying the viewer to incorporate alternative color representation methods

### ISSUE THREAD

**Issue by JIANG-CX** [graphdeco-inria/gaussian-splatting#75]
Hi, I noticed that the real-time viewer currently only supports SH. Could you please help me in modifying the viewer to incorporate alternative color representation methods, such as an MLP, to obtain the color information for each Gaussian? I would greatly appreciate your guidance in this matter. 

Additionally, I have noticed that only one image is used for optimization in each iteration. I am curious to know if the rasterizer supports batch operations that would allow rendering a batch of images for training

Thank you.

**Comment by Snosixtyboo** [comment]
Hi,

sorry, we think that would be a bigger change to the technique, so we won't be able to give detailed guidance. BUT, there is the option to override the colors for the rasterizer, meaning, you can add your own solution to compute RGB from attributes with Pytorch. These can then be passed in (see the ```gaussian_renderer``` code). You can follow the path of the renderer that precomputes the RGB from SHs via PyTorch:

```
    colors_precomp = None
    if colors_precomp is None:
        if pipe.convert_SHs_python:
        ...
```

You can replace what comes after this with your own solution to produce RGB from optimized parameters, e.g., by sending them through a small MLP. The gradients will then be taken care of by autodiff. Note that this does not yet give you a solution for viewing results in the real-time viewer after training has completed.

Hth,
Bernhard
